src/Symfony/Bridge/Doctrine/SchemaListener/AbstractSchemaListener.php

```diff
@@ -59,13 +59,9 @@ protected function getIsSameDatabaseChecker(Connection $connection): \Closure
             }
 
             try {
-                $rowCount = $connection->executeStatement('DELETE FROM _schema_subscriber_check WHERE key == :key', ['key' => $key], ['key' => Types::STRING]);
-
-                return 0 === $rowCount;
+                return !$connection->executeStatement('DELETE FROM _schema_subscriber_check WHERE key == :key', ['key' => $key], ['key' => Types::STRING]);
             } finally {
-                [$count] = $connection->executeQuery('SELECT count(id) FROM _schema_subscriber_check')->fetchOne();
-
-                if (!$count) {
+                if (!$connection->executeQuery('SELECT count(id) FROM _schema_subscriber_check')->fetchOne()) {
                     try {
                         $schemaManager->dropTable('_schema_subscriber_check');
```

Comment:
I now realize that this DROP TABLE creates a race condition, isn't it?

Reply:
Yes, a race condition is possible when dropping this table.